packages/iov-bns-governance/src/governor.ts

```diff
@@ -0,0 +1,111 @@
+import BN = require("bn.js");
+
+import { Address, Identity, WithCreator } from "@iov/bcp";
+import {
+  bnsCodec,
+  BnsConnection,
+  BnsTx,
+  CreateProposalTx,
+  Electorate,
+  Proposal,
+  TallyTx,
+  VoteOption,
+  VoteTx,
+} from "@iov/bns";
+import { Encoding } from "@iov/encoding";
+
+import { CommitteeId, ProposalOptions, ProposalType } from "./proposals";
+
+/* Constants to be finalised once the committees are actually on-chain */
+export const COMMITTEE_IDS = {
+  testnet: {
+    GOVERNING_BOARD: 1 as CommitteeId,
+    ECONOMIC_COMMITTEE: 2 as CommitteeId,
+    VALIDATOR_COMMITTEE: 3 as CommitteeId,
+    TECH_COMMITTEE: 4 as CommitteeId,
+  },
+};
+
+export interface GovernorOptions {
+  readonly connection: BnsConnection;
+  readonly identity: Identity;
+}
+
+export class Governor {
+  private readonly connection: BnsConnection;
+  private readonly identity: Identity;
+  private readonly address: Address;
+
+  public constructor({ connection, identity }: GovernorOptions) {
+    this.connection = connection;
+    this.identity = identity;
+    this.address = bnsCodec.identityToAddress(this.identity);
+  }
+
+  public async getElectorates(): Promise<readonly Electorate[]> {
+    const electorates = await this.connection.getElectorates();
+    return electorates.filter(({ electors }) =>
+      Object.keys(electors).some(key => key.toUpperCase() === this.address.toUpperCase()),
```

Comment:
BNS addresses have a deterministic casing and as soon as an `Address` instance is created, we can assume the casing is correct. User input needs to be normalized in the place where the input happens.

Reply:
Perfect, wasn't sure about this.